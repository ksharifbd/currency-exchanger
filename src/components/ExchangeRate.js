import React from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import styled from '@material-ui/core/styles/styled';

const ExchangeRate = ({ from, to }) => {
  const USDOLLAR_UNICODE = 36;
  const GBPOUND_UNICODE = 163;
  const EURO_UNICODE = 8364;

  const getCurrencyCode = currency => {
    switch (currency) {
      case 'GBP':
        return GBPOUND_UNICODE;
      case 'EUR':
        return EURO_UNICODE;
      default:
        return USDOLLAR_UNICODE;
    }
  };

  const getCurrencySign = currency => {
    return String.fromCharCode(getCurrencyCode(currency));
  };

  const ExchangeRateWrapper = styled(Box)({
    display: 'flex',
    justifyContent: 'center',
  });

  return (
    <ExchangeRateWrapper mb={3}>
      <Box>
        <Box component="h2">Rate</Box>
        <Box component="span">
          {getCurrencySign(from.currency)}
          {from.value} = {getCurrencySign(to.currency)}
          {to.value}
        </Box>
      </Box>
    </ExchangeRateWrapper>
  );
};

ExchangeRate.propTypes = {
  from: PropTypes.shape({
    currency: PropTypes.oneOf(['USD', 'EUR', 'GBP']),
    value: PropTypes.number,
  }),
  to: PropTypes.shape({
    currency: PropTypes.oneOf(['USD', 'EUR', 'GBP']),
    value: PropTypes.number,
  }),
};

ExchangeRate.defaultProps = {
  from: {
    currency: 'EUR',
    value: 1,
  },
  to: {
    currency: 'GBP',
    value: 1,
  },
};

export default ExchangeRate;
