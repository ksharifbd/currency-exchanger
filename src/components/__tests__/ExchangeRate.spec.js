import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import ExchangeRate from '../ExchangeRate';

describe('Exchange Rate', () => {
  let mount;

  beforeAll(() => {
    mount = createMount();
  });

  afterAll(() => {
    mount.cleanUp();
  });

  const tree = (from, to) => {
    return mount(<ExchangeRate from={from} to={to} />);
  };

  it('should return the default setup correctly', () => {
    expect(tree()).toMatchSnapshot();
  });

  it('should render with correct symbol', () => {
    expect(
      tree({ currency: 'USD', value: 1 })
        .find('span')
        .text()
    ).toEqual(expect.stringContaining('$'));
  });
});
