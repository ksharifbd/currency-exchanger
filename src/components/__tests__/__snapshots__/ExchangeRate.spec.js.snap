// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange Rate should return the default setup correctly 1`] = `
<ExchangeRate
  from={
    Object {
      "currency": "EUR",
      "value": 1,
    }
  }
  to={
    Object {
      "currency": "GBP",
      "value": 1,
    }
  }
>
  <Styled(Styled(MuiBox))
    mb={3}
  >
    <Styled(MuiBox)
      className="Styled(MuiBox)-root-1"
      mb={3}
    >
      <div
        className="MuiBox-root MuiBox-root-3 Styled(MuiBox)-root-1"
      >
        <Styled(Styled(MuiBox))>
          <Styled(MuiBox)
            className="Styled(MuiBox)-root-4"
          >
            <div
              className="MuiBox-root MuiBox-root-5 Styled(MuiBox)-root-4"
            >
              <Styled(MuiBox)
                component="h2"
              >
                <h2
                  className="MuiBox-root MuiBox-root-6"
                >
                  Exchange Rate
                </h2>
              </Styled(MuiBox)>
              <Styled(MuiBox)
                component="span"
              >
                <span
                  className="MuiBox-root MuiBox-root-7"
                >
                  €
                  1
                   = 
                  £
                  1
                </span>
              </Styled(MuiBox)>
            </div>
          </Styled(MuiBox)>
        </Styled(Styled(MuiBox))>
      </div>
    </Styled(MuiBox)>
  </Styled(Styled(MuiBox))>
</ExchangeRate>
`;
